Source/Core/DolphinQt/Config/HardcoreWarningWidget.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include "DolphinQt/Config/HardcoreWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+
+#include "Core/Config/AchievementSettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+
+#include "DolphinQt/Settings.h"
+
+HardcoreWarningWidget::HardcoreWarningWidget(QWidget* parent) : QWidget(parent)
+{
+  CreateWidgets();
+
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this, [this] { Update(); });
+
+  Update();
+}
+
+void HardcoreWarningWidget::CreateWidgets()
+{
+  auto* icon = new QLabel;
+
+  const auto size = 1.5 * QFontMetrics(font()).height();
+
+  QPixmap warning_icon = style()->standardIcon(QStyle::SP_MessageBoxWarning).pixmap(size, size);
+
+  icon->setPixmap(warning_icon);
+
+  m_text = new QLabel(tr("This feature is disabled in hardcore mode."));
```

Comment:
This is very unclear without any further context for a user who may not remember what 'hardcore mode' is. You should maybe take a page out of the `CheatWarningWidget` and add a button that links to the Achievements widget.

Reply:
I have *mostly* implemented this but continue to have two issues I'm currently investigating. One, the button in FreeLook doesn't open the achievements dialog like it ought to. Two, the Riivolution widget insists on constantly taking focus, even over the achievements dialog, which looks really wonky.

Reply:
The Riivolution widget inherits from QDialog, which is modal by default and usually intended for dialogs that require you to handle them before doing anything else. If you're trying to open the Achievements dialog from there you'll want to close the Riivolution widget too so the user can interact with the Achievements dialog.

Reply:
FreeLook is now working.